mypy/server/deps.py

```diff
@@ -101,14 +154,99 @@ def visit_import(self, o: Import) -> None:
+        if isinstance(lvalue, IndexExpr):
+            self.add_indexing_method_dependency(lvalue, lvalue=True)
+        elif isinstance(lvalue, NameExpr):
+            if lvalue.kind in (MDEF, GDEF):
+                # Assignment to an attribute in the class body, or direct assignment to a
+                # global variable.
+                lvalue_type = self.type_map[lvalue]
+                type_triggers = get_type_dependencies(lvalue_type)
+                attr_trigger = make_trigger('%s.%s' % (self.target_stack[-1], lvalue.name))
+                for type_trigger in type_triggers:
+                    self.add_dependency(type_trigger, attr_trigger)
+        elif isinstance(lvalue, MemberExpr):
+            if lvalue.kind is None:
+                # Reference to a non-module attribute
+                object_type = self.type_map[lvalue.expr]
+                lvalue_type = self.type_map[lvalue]
+                type_triggers = get_type_dependencies(lvalue_type)
+                for attr_trigger in self.attribute_triggers(object_type, lvalue.name):
+                    for type_trigger in type_triggers:
+                        self.add_dependency(type_trigger, attr_trigger)
+        elif isinstance(lvalue, (ListExpr, TupleExpr)):
+            for item in lvalue.items:
+                self.process_lvalue(item)
+        # TODO: star lvalue
+
+    def visit_operator_assignment_stmt(self, o: OperatorAssignmentStmt) -> None:
+        super().visit_operator_assignment_stmt(o)
+        self.process_lvalue(o.lvalue)
+        method = op_methods[o.op]
+        self.add_attribute_dependency_for_expr(o.lvalue, method)
+        if o.op in ops_with_inplace_method:
+            inplace_method = '__i' + method[2:]
+            self.add_attribute_dependency_for_expr(o.lvalue, inplace_method)
+
+    def visit_for_stmt(self, o: ForStmt) -> None:
+        super().visit_for_stmt(o)
+        self.add_attribute_dependency_for_expr(o.expr, '__iter__')
+        self.process_lvalue(o.index)
+        if isinstance(o.index, (TupleExpr, ListExpr)):
+            # Process multiple assignment to index variables.
+            item_type = o.inferred_item_type
+            if item_type:
+                self.add_attribute_dependency(item_type, '__iter__')
+        if o.index_type:
+            self.add_type_dependencies(o.index_type)
+
+    def visit_with_stmt(self, o: WithStmt) -> None:
+        super().visit_with_stmt(o)
+        for e in o.expr:
+            self.add_attribute_dependency_for_expr(e, '__enter__')
+            self.add_attribute_dependency_for_expr(e, '__exit__')
+        if o.target_type:
+            self.add_type_dependencies(o.target_type)
+
+    def visit_print_stmt(self, o: PrintStmt) -> None:
+        super().visit_print_stmt(o)
+        if o.target:
+            self.add_attribute_dependency_for_expr(o.target, 'write')
+
+    def visit_del_stmt(self, o: DelStmt) -> None:
+        super().visit_del_stmt(o)
+        if isinstance(o.expr, IndexExpr):
```

Comment:
Could also be deleting an attribute (del x.y), should depend on that then.

Reply:
This was working already due to the `super()` traversal method.